contract/src/main/kotlin/com/ritense/valtimo/contract/security/config/oauth2/NoOAuth2ClientsConfiguredCondition.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ritense.valtimo.contract.security.config.oauth2
+
+import org.springframework.boot.autoconfigure.condition.ConditionMessage
+import org.springframework.boot.autoconfigure.condition.ConditionOutcome
+import org.springframework.boot.autoconfigure.condition.SpringBootCondition
+import org.springframework.boot.autoconfigure.security.oauth2.client.OAuth2ClientProperties
+import org.springframework.boot.context.properties.bind.Bindable
+import org.springframework.boot.context.properties.bind.Binder
+import org.springframework.context.annotation.ConditionContext
+import org.springframework.core.env.Environment
+import org.springframework.core.type.AnnotatedTypeMetadata
+
+// The opposite of: org.springframework.boot.autoconfigure.security.oauth2.client.ClientsConfiguredCondition
```

Comment:
That class states `Condition that matches if any spring.security.oauth2.client.registration properties are defined.`

It does not explain how it works, of why we need the negated version of it.

Reply:
When an application has configured the `spring.security.oauth2.client.registration` property, Valtimo will switch out its old authentication services/bean to use then new OAuth2 authentication beans.